server_modules/file_system.js

```diff
@@ -77,24 +75,21 @@ exports.route = function(app) {
         var file = exports.getDataFilePath(req.params[0]);
         fs.stat(file, function(err, stat) {
             if (err) {
-                log.error("file not found {0}".format(file));
-                res.status(404).send('file not found');
+                utils.sendError({ status: 404, message: `file not found ${file}` })
```

Comment:
I don't think any of these `sendError()` calls do what you want. Did you test the error scenarios?